src/main/java/com/minecolonies/coremod/colony/buildings/BuildingMiner.java

```diff
@@ -637,10 +636,32 @@ public void setClearedShaft(final boolean clearedShaft)
         this.clearedShaft = clearedShaft;
     }
 
+    @Override
+    public void searchWorkOrder()
+    {
+        final CitizenData citizen = getMainWorker();
+        if (citizen == null)
+        {
+            return;
+        }
+
+        final List<WorkOrderBuildMiner> list = getColony().getWorkManager().getOrderedList(WorkOrderBuildMiner.class);
+
+        for (final WorkOrderBuildMiner wo: list)
+        {
+            if(this.getID().equals(wo.getMinerBuilding()))
+            {
+                citizen.getJob(JobMiner.class).setWorkOrder(wo);
+                wo.setClaimedBy(citizen);
+                return;
+            }
+        }
+    }
+
     /**
      * Provides a view of the miner building class.
      */
-    public static class View extends AbstractBuildingWorker.View
+    public static class View extends AbstractBuildingBuilderView
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this class from 39 to the maximum allowed 30 or externalize it in a public class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)